packages/plugin/package.json

```diff
@@ -2,13 +2,19 @@
   "name": "@theia/plugin",
   "version": "0.3.8",
   "description": "Theia - Plugin API",
-  "types": "./src/theia.d.ts",
+  "main": "lib/common/index.js",
```

Comment:
I think it would be time to split this module in two because clients are using @theia/plugin to get typings, etc
while here we're adding "server-side/extension implementation" dependencies

so maybe we should have packages/plugin-shared (for typescript), and packages/plugin (for the clients like command line utilities in addition to plugin-api, etc) and packages/plugin-extension (for the extension about plugins)

Reply:
I exposed typings for Che specific extension which contribute to plugin extension. It is not supposed to be extended by a plugin.

I agree we have to separate typings for users plugins to not to depends on Theia, but I think that is another issue

Reply:
the issue is that by adding
```
 "dependencies": {
+    "@theia/core": "^0.3.8",
+    "@theia/filesystem": "^0.3.8"
```

with that, when you'll create a new plugin or trying the samples, you'll download tons of dependencies including some electron native compiling, etc, slowing down the scaffolding

Reply:
I haven't seen separate module for declarations yet, do we have it?
As about dependencies, they aren't supposed to be exported to plugins. Separation of plugin extension implementation and declarations for plugins should do the job, if I understand you correctly.

Reply:
I will do changes in a new PR, it will be faster.